ingestion/src/metadata/ingestion/source/database/common_db_source.py

```diff
@@ -697,7 +700,49 @@ def yield_view_lineage(self) -> Iterable[Either[OMetaLineageRequest]]:
         else:
             yield from self._process_view_def_serial()
 
-    def _get_foreign_constraints(self, foreign_columns) -> List[TableConstraint]:
+    def _is_column_unique(self, column: Dict, columns: List[Column]) -> bool:
+        """
+        Method to check if the column in unique in the table
+        """
+        if column and len(column) > 0:
+            constrained_column = column[0]
+            for col in columns or []:
+                if model_str(col.name) == constrained_column:
+                    if col.constraint and col.constraint.value in {
+                        ConstraintType.UNIQUE.value,
+                        ConstraintType.PRIMARY_KEY.value,
+                    }:
+                        return True
+                    break
+        return False
+
+    def _get_relationship_type(
+        self, column: Dict, referred_table_columns: List[Column], columns: List[Column]
```

Comment:
same as above. We can have a `constraints` module that takes care of these things and for us to test